sample/shared/compose/src/commonMain/kotlin/com/arkivanov/sample/shared/root/RootContent.kt

```diff
@@ -109,6 +124,8 @@ fun RootContent(component: RootComponent, modifier: Modifier = Modifier) {
                 label = { Text(text = "Custom Nav", softWrap = false) },
             )
         }
+
+        DialogContent(dialogComponent = dialogState.overlay?.instance)
```

Comment:
You can guard this like:

```
dialogState?.overlay?.instance?.let {
    DialogContent(it)
}
``

This way the `DialogContent` can have non-optional component

Reply:
I'm also looking for some more simple and clean way to simplify such checks, any hints @arkivanov?

Reply:
done

Reply:
Nope, `dialogState?.overlay?.instance?.also {` looks like a cleanest way.